src/datasources/config-api/config-api.service.ts

```diff
@@ -13,20 +16,38 @@ export class ConfigApi {
     private readonly dataSource: CacheFirstDataSource,
     @Inject(IConfigurationService)
     private readonly configurationService: IConfigurationService,
+    private readonly validationErrorFactory: ValidationErrorFactory,
   ) {
     this.baseUri =
       this.configurationService.getOrThrow<string>('safeConfig.baseUri');
   }
 
   async getChains(): Promise<Page<Chain>> {
     const key = 'chains'; // TODO key is not final
-    const url = this.baseUri + '/api/v1/chains';
-    return await this.dataSource.get(key, url);
+    const url = `${this.baseUri}/api/v1/chains`;
+    const page: Page<Chain> = await this.dataSource.get(key, url);
+    const chains = page.results;
+
+    if (!chains.every((chain) => isValidChain(chain))) {
```

Comment:
I took this approach where `Page` is not validated explicitly and the page contents are. This is an opinionated thing, other option could be create multiple `PageWhateverSchema` and validate the `Page` as well. I don't have a lot of expertise in JSONSchema definitions, maybe there's a way to compose the validation schema dynamically 🤔

Reply:
I think having a schema for `Page` can be useful as it allow also to validate incoming/outgoing urls for the `next` and `prev`.

We can tackle this on a different PR 👍

Reply:
You mean running explicitly two validations, one against `Page` schema, and another one against the `Item` schema which is wrapped into the `Page`? Or composing schemas in advance? For sure we can tackle this in a new issue/PR, I can create one 👍